src/main/java/org/osc/controller/nsc/restserver/api/InspectionApis.java

```diff
@@ -0,0 +1,403 @@
+import org.osgi.service.jdbc.DataSourceFactory;
+import org.osgi.service.jpa.EntityManagerFactoryBuilder;
+import org.osgi.service.transaction.control.TransactionControl;
+import org.osgi.service.transaction.control.jpa.JPAEntityManagerProviderFactory;
+
+@Component(service = InspectionApis.class)
+@Path("/nw")
+@Consumes({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+@Produces({ MediaType.APPLICATION_JSON, MediaType.APPLICATION_XML })
+public class InspectionApis {
+    private static Logger LOG = Logger.getLogger(InspectionApis.class);
+    private EntityManager em;
+    private RedirectionApiUtils utils;
+    private SampleSdnRedirectionApi api;
+    @Reference(target = "(osgi.local.enabled=true)")
+    private TransactionControl txControl;
+    @Reference(target = "(osgi.unit.name=nsc-mgr)")
+    private EntityManagerFactoryBuilder builder;
+    @Reference(target = "(osgi.jdbc.driver.class=org.h2.Driver)")
+    private DataSourceFactory jdbcFactory;
+    @Reference(target = "(osgi.local.enabled=true)")
+    private JPAEntityManagerProviderFactory resourceFactory;
+    private String ipAddr;
+
+    public void init(String ipAddr) throws Exception {
+        if (ipAddr != null) {
+            if (ipAddr.equals(this.ipAddr)) {
+                return;
+            }
+        } else {
+            return;
+        }
+
+        Properties props = new Properties();
+        props.setProperty(JDBC_URL,
+                "jdbc:h2:./nscPlugin_" + ipAddr + ";MVCC\\=TRUE;LOCK_TIMEOUT\\=10000;MV_STORE=FALSE;");
+        props.setProperty(JDBC_USER, "admin");
+        props.setProperty(JDBC_PASSWORD, "admin123");
+        DataSource ds = null;
+        try {
+            ds = this.jdbcFactory.createDataSource(props);
+        } catch (SQLException error) {
+            LOG.error(error);
+            throw new IllegalStateException(error.getMessage(), error);
+        }
+        EntityManager em = this.resourceFactory
+                .getProviderFor(this.builder, singletonMap("javax.persistence.nonJtaDataSource", (Object) ds), null)
+                .getResource(this.txControl);
+        this.em = em;
+        this.api = new SampleSdnRedirectionApi(this.txControl, this.em);
+        this.utils = new RedirectionApiUtils(this.em, this.txControl);
+        this.ipAddr = ipAddr;
+
+    }
+
+    @Path("/ipAddr/{ipAddr}/netelem")
+    @POST
+    public String createNetworkElement(@PathParam("ipAddr") String ipAddr, NetworkElementEntity entity)
+            throws Exception {
+        if (entity == null) {
+            throw new IllegalArgumentException("Attempt to create network element with no network entities");
+        }
+        LOG.info(String.format("Creating network element with elementId  id %s)", "" + entity.getElementId()));
+
+        init(ipAddr);
+        return this.txControl.required(() -> {
+            NetworkElementEntity element = this.utils.txNetworkElementEntityByElementId(entity.getElementId());
```

Comment:
Typically the id wouldnt be provided if you are doing a create so instead if the user is passing an Id in the request/entity you should probably throug an illegalargumentexception. You may repeat the same pattern for the other apis

Reply:
If I look at the NetworkElementEntity, id value is not generated automatically as annotation generated value is not used. so user passing the value is valid here.
@dmitryintel , please comment.